packages/realm-react/src/cachedCollection.ts

```diff
@@ -39,6 +43,12 @@ type CachedCollectionArgs<T> = {
    * Callback which is called whenever an object in the collection changes
    */
   updateCallback: () => void;
+
+  /**
+   * Reference boolean which is set to true whenever an object in the collection changes
+   */
+  updateRef: React.MutableRefObject<boolean> | UpdateCollectionRef;
```

Comment:
It's a bit of a hack, but for lists on an object, I also needed a way to create a stable reference, so I just reused the structure of a `React.MutableRefObject`, since the lists cache instantiation is out of the React context (can't call `useRef`).  I can try to see if just having a similar shape to the type passes the type check.

Reply:
Oh nice, i guess the interface of `useRef` is just:
```
{
current: T
}
```
so I just need to provide something in that shape to use it.